cordapp-example/workflows-kotlin/src/main/kotlin/com/example/utils/IouStateFinder.kt

```diff
@@ -0,0 +1,20 @@
+package com.example.utils
+
+import com.example.state.IOUState
+import net.corda.core.contracts.StateAndRef
+import net.corda.core.node.ServiceHub
+import net.corda.core.node.services.vault.QueryCriteria
+import java.util.*
+
+object IouStateFinder {
+    fun call(serviceHub: ServiceHub, iouStateLinearId: String): StateAndRef<IOUState> {
```

Comment:
Have you thought of trying this function with a receiver? Not even in a class or object.

```kotlin
fun ServiceHub.findIOUState(iouStateLinearId: String): StateAndRef<IOUState> {
```

Then, when you call it, it is `serviceHub.findIOUState(iouStateLinearId)`.

Reply:
done... what a nice feature :)